src/sentry/receivers/tokens.py

```diff
@@ -0,0 +1,17 @@
+from django.db.models.signals import pre_save
+from django.dispatch import receiver
+
+from sentry.conf.server import SENTRY_SCOPE_HIERARCHY_MAPPING, SENTRY_SCOPES
+from sentry.models import ApiKey, ApiToken
+
+
+@receiver(pre_save, sender=ApiKey, dispatch_uid="enforce_scope_hierarchy_api_key")
+@receiver(pre_save, sender=ApiToken, dispatch_uid="enforce_scope_hierarchy_api_token")
+def enforce_scope_hierarchy(instance, **kwargs) -> None:
```

Comment:
should you be doing this every time somebody saves a token, or only on token creation? i ask because you will be running a script to update existing tokens anyway

Reply:
For org/integration tokens, you can still update scopes after the token has been created. If you do it through the UI it should automatically have hierarchy, but it's possible to update it through the API and circumvent that so we still need it on save.